src/mem/alloc.h

```diff
@@ -6,6 +6,36 @@
 #  define ALLOCATOR
 #endif
 
+#ifdef SNMALLOC_PASS_THROUGH
+#  include <cstdlib>
+#  if defined(_WIN32) || defined(__APPLE__)
+#    error "Pass through not supported on this platform"
+//   The Windows aligned allocation API is not capable of supporting the
+//   snmalloc API Apple was not providing aligned memory in some tests.
+#  else
+#    define ALIGNED_ALLOC std::aligned_alloc
```

Comment:
Also, note that `std::aligned_alloc` is just a C++ namespacing of the C11 `aligned_alloc`, which isn't very well supported.  On everything that's not Windows, `posix_memalign` is probably more likely to exist and work (it's from POSIX2001, so people have had 10 more years to get around to implementing it); however, POSIX does not provide a way to free memory allocated with `posix_memalign` (glibc allows it to be freed with `free`) so the whole thing is a bit of a mess.

Reply:
Will switching to `using` have the desired effect? If I do a direct replacement, it is an unnamespaced `using` so will affect all future files like a `#define`.  I can put it inside a namespace, but the includes will be outside the namespace, so I will be doubling up the `#ifdef` testing.

I can put the SNMALLOC prefix on, but it is kind of weird as they aren't snmalloc.  Perhaps
```C++
NOT_SNMALLOC_MSIZE
```
makes the most sense?

Reply:
If they're in the `snmalloc` namespace, then there's no pollution with `using`.  I'd generally prefer to have any macros defined by snmalloc start with `SNMALLOC` because no other library should be using that prefix by accident.  Having them start with `NOT_` is a bit weird: they're not snmalloc functions, but they are snmalloc macro definitions.

Reply:
So 
```C++
# elif defined(__FreeBSD__)
#      include <malloc_np.h>
#      using MSIZE = malloc_usable_size;
#    else
```
if this is inside a namespace, then that namespace will apply to `malloc_np.h`, which will probably cause issues?  And otherwise they are global.  So it means duplicating the `#ifdef`, both inside and outside the namespace.

Reply:
Alternatively you could write it as:

```
#    elif defined(__FreeBSD__)
#      include <malloc_np.h>
namespace snmalloc { namespace external_malloc {
        using malloc_size = malloc_usable_size;
}}
#    else
```
You could then `using namespace snmalloc::external_malloc` or similar in the `Alloc` class to import these.